### PR TITLE
chore: update documentation for commits and promotion 

### PR DESCRIPTION
**Pull request by cellomatt**
This PR updates the Environments concept page in our docs and moves the concepts of Commits and Promotion to a separate page. The changes include:

- Rearrange the order of some content to make it clearer
- Add additional search tags
- Add a new section on commit diffs (including a gif asset)
- Add explicit instructions on how to take certain actions in your dashboard, and mention the ability to perform these same actions via the MAPI or CLI
- Add Commits to sidebar nav
- Add Commits to the Concepts overview page, re-order the concepts on the overview page to match the nav ordering, and add a section for Recipients, which was missing
- Update the Workflows concept page's backlinks to separate version control and environments

### CODE REVIEW

**Review by cjbell — COMMENTED** [review]
All of these changes in here look really great and I think really help explain the concept, but this makes me wonder: should we have commits as it's own page, or maybe we should call this page "Environments & commits". WDYT?

File: content/concepts/environments.mdx [hunk 1]
```diff
@@ -23,12 +30,20 @@ The API key you use determines the environment into which you'll be sending data
...
-A couple other things to note:
+**A few things to note:**
 
 - Channel configurations, branding, and variables do not need to be committed, as they live at the account-level. This means that if you make a change to a channel configuration, it will update immediately on notifications sent in that environment.
-- Any changes you have not yet committed to your current environment **will** apply when you're using the test runner. This way you can test your latest changes before you commit them to your environment.
+- Any changes you have saved but not yet committed **will** apply when you're using the test runner. This way you can test your latest changes before you commit them to your development environment.
+- You can work with Knock resources outside of your dashboard if you prefer. We offer both a [Management API](/developer-tools/management-api) and a [command line interface](/developer-tools/knock-cli) for interacting with Knock resources programmatically.
+- The commit model applies to all methods of interacting with Knock resources, whether directly in the dashboard or with the Management API and CLI.
+
+### Commit Diffs
```
Comment (samseely) [inline content/concepts/environments.mdx:42]:
```suggestion
### Commit diffs
```

Nit: this should be sentence case not title case

**Comment by samseely** [comment]
> All of these changes in here look really great and I think really help explain the concept, but this makes me wonder: should we have commits as it's own page, or maybe we should call this page "Environments & commits". WDYT?

Yep agreed with this note from @cjbell . I'd lean towards "Commits" as its own page to keep as a clean separate concept that users can easily find. That page could also do a deeper dive walkthrough of the version control model in Knock. We'd probably move promotion/rollback into that (which makes sense give revert commit is coming out soon). When we get into branching that would be a good place for that information too.

**Comment by cellomatt** [comment]
> > All of these changes in here look really great and I think really help explain the concept, but this makes me wonder: should we have commits as it's own page, or maybe we should call this page "Environments & commits". WDYT?
> 
> Yep agreed with this note from @cjbell . I'd lean towards "Commits" as its own page to keep as a clean separate concept that users can easily find. That page could also do a deeper dive walkthrough of the version control model in Knock. We'd probably move promotion/rollback into that (which makes sense give revert commit is coming out soon). When we get into branching that would be a good place for that information too.

Ha, I was having this same conversation with @rachael-t! Sounds good, will make that change 😄 

image

File: content/concepts/commits.mdx [hunk 2]
```diff
@@ -0,0 +1,70 @@
+---
+title: Commits
```
Comment (cjbell) [inline content/concepts/commits.mdx:2]: Now that this is a separate page, we also need to add to `data/sidebar.ts` under "Concepts" so it appears as a nav item in the sidebar

**Review by cjbell — APPROVED** [review]
one last comment but looking great

File: content/concepts/commits.mdx (hunk 2 above)
Comment (cellomatt) [reply content/concepts/commits.mdx:2]: So weird, I had this in my local version and somehow it didn't get tracked 🤔 anyway, added, thanks!

**Pull request merged**